src/engine/GroupBy.cpp

```diff
@@ -764,16 +791,23 @@ void GroupBy::computeResult(ResultTable* result) {
 
   int inWidth = subresult->_data.cols();
   int outWidth = result->_data.cols();
-  CALL_FIXED_SIZE_2(inWidth, outWidth, doGroupBy, subresult->_data,
-                    inputResultTypes, groupByCols, aggregates, &result->_data,
-                    subresult.get(), result, getIndex());
-
-  // Free the user data used by GROUP_CONCAT aggregates.
-  for (Aggregate& a : aggregates) {
-    if (a._type == ParsedQuery::AggregateType::GROUP_CONCAT) {
-      delete static_cast<std::string*>(a._userData);
+
+  auto cleanup = [&]() {
+    // Free the user data used by GROUP_CONCAT aggregates.
+    for (Aggregate& a : aggregates) {
+      if (a._type == ParsedQuery::AggregateType::GROUP_CONCAT) {
+        delete static_cast<std::string*>(a._userData);
+      }
```

Comment:
TODO: Write issue etc. that the `_userData` becomes a type with a proper destructor.